Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -155,6 +155,12 @@ public override void Analyze(AnalyzeContext context)
             {
                 try
                 {
+                    var fileInfo = new FileInfo(filePath);
+                    if (context.FileSizeInKilobytes != -1 && fileInfo.Length / 1024 > context.FileSizeInKilobytes)
```

Comment:
>fileInfo [](http://example.com/codeflow?start=61&length=8)

There are many corner cases you aren't covering, in part because you've taken the route of using simple checked in files for your tests. Instead, you could have created a unit test strategy that allows you to mock arbitrary file sizes so you can test them with arbitrary file size in kilobyte settings.

The trick, of course, is finding the right layer to wrap/inject your testing.

Reply:
We're now covering some of these with already established tests. VS is throwing an exception trying to run code coverage for SPAM. It looks like there may be a custom test setting that conflicts with a resource VS requires to run the analysis.